### PR TITLE
Resize helper

### PR DESCRIPTION
**Pull request by Tim-Anderson**
Update resize-helper to remove parted and replace with fdisk.

This removes any use of gplv3 content and so it can be included with gplv2 only builds.

### CODE REVIEW

File: meta-lmp-base/recipes-support/resize-helper/resize-helper_0.1.bb [hunk 1]
```diff
@@ -5,7 +5,7 @@ LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/BSD-2-Clause;md5=cb641bc04cda31
 
 inherit allarch systemd
 
-RDEPENDS:${PN} += "e2fsprogs-resize2fs gptfdisk parted util-linux-findmnt"
+RDEPENDS_${PN} += "e2fsprogs-resize2fs gptfdisk util-linux-findmnt"
```
Comment (angolini) [inline meta-lmp-base/recipes-support/resize-helper/resize-helper_0.1.bb:8]: For main, it's `RDEPENDS:${PN}`

**Review by angolini — APPROVED** [review]
We probably want this also on hardknott
So, this is a reminder about the override syntax being different for main and hardknott

**Comment by Tim-Anderson** [comment]
@Ricardo Salveti ***@***.***> which targets use gpt
partition?

**Comment by ricardosalveti** [comment]
> @ricardo Salveti ***@***.***> which targets use gpt partition?

Easier one to test is intel.

```
meta-lmp-base/wic/efidisk-sota.wks:bootloader --source bootimg-sota-efi --ptable gpt --configfile="grub-ota.cfg"
meta-lmp-base/wic/image-efi-installer.wks.in:bootloader --ptable gpt --timeout=5
meta-lmp-bsp/wic/sdimage-freedom-u540-spl-sota.wks:bootloader --ptable gpt
meta-lmp-bsp/wic/sdimage-freedom-u540-spl.wks:bootloader --ptable gpt
meta-lmp-bsp/wic/sdimage-stm32mp157c-dk2-trusted-sota.wks.in:bootloader --ptable gpt
```

File: meta-lmp-base/recipes-support/resize-helper/resize-helper_0.1.bb (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-base/recipes-support/resize-helper/resize-helper_0.1.bb:8]: @Tim-Anderson syntax will need to be updated.

Comment (Tim-Anderson) [reply meta-lmp-base/recipes-support/resize-helper/resize-helper_0.1.bb]: corrected

**Review by MrCry0 — APPROVED** [review]
LGTM

File: meta-lmp-base/recipes-support/resize-helper/resize-helper/resize-helper [hunk 2]
```diff
@@ -23,40 +23,48 @@
...
-PART_TABLE_TYPE=$(udevadm info --query=property --name=${ROOT_DEVICE} | grep '^ID_PART_TABLE_TYPE=' | cut -d'=' -f2)
+PART_TABLE_TYPE=$(echo "${INFO}" | grep '^ID_PART_TABLE_TYPE=' | cut -d'=' -f2)
 # find the block device
 DEVICE=$(udevadm info --query=path --name=${ROOT_DEVICE} | awk -F'/' '{print $(NF-1)}')
 DEVICE="/dev/${DEVICE}"
 
+SIZE=$(blockdev --getsz ${DEVICE})
 if [ "$PART_TABLE_TYPE" = "gpt" ]; then
 	${SGDISK} -e ${DEVICE}
-	${PARTPROBE}
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-support/resize-helper/resize-helper/resize-helper:55]: At this point the backup GPT data structure will be at the end of the disk, which I believe will change the partition table as a consequence, and why we had partprobe being called here before.

I think it is better to have partprobe still being called, and for that we can enable CONFIG_PARTPROBE in busybox and use the tool from it instead.

Same after resizing the partition.

File: meta-lmp-base/recipes-support/resize-helper/resize-helper_0.1.bb [hunk 3]
```diff
@@ -5,7 +5,7 @@ LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/BSD-2-Clause;md5=cb641bc04cda31
 
 inherit allarch systemd
 
-RDEPENDS:${PN} += "e2fsprogs-resize2fs gptfdisk parted util-linux-findmnt"
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-support/resize-helper/resize-helper_0.1.bb:8]: Please rdepend on util-linux-fdisk and util-linux-blockdev, which are now required.

File: meta-lmp-base/recipes-support/resize-helper/resize-helper/resize-helper [hunk 4]
```diff
@@ -23,40 +23,48 @@
...
 fi
 
-# 'Yes' is required when partition is already mounted
-echo -e 'Yes\n100%' | ${PARTED} ---pretend-input-tty ${DEVICE} resizepart ${PART_ENTRY_NUMBER} ||
-	echo -e '100%' | ${PARTED} ---pretend-input-tty ${DEVICE} resizepart ${PART_ENTRY_NUMBER}
-${PARTPROBE}
+# Use fdisk to repartition
+END=$((${SIZE} - 1))
+PARTOF=$(echo "${INFO}" | grep '^ID_PART_ENTRY_OFFSET=' | cut -d'=' -f2)
+echo -e "d\n${PART_ENTRY_NUMBER}\nn\np\n${PART_ENTRY_NUMBER}\n${PARTOF}\n${END}\nw\n" | ${FDISK} ${DEVICE} > /dev/null
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-support/resize-helper/resize-helper/resize-helper:67]: Is it so verbose that it is better to dump to /dev/null? I would prefer for it to go to stdout as we can then see what happened via journal.

Comment (ricardosalveti) [reply meta-lmp-base/recipes-support/resize-helper/resize-helper/resize-helper:67]: Please add a comment as well saying what it is being done here (deleting partition and creating a new one up to END)

Comment (ricardosalveti) [reply meta-lmp-base/recipes-support/resize-helper/resize-helper/resize-helper:67]: And we do need partprobe after this gets called.

Comment (ricardosalveti) [reply meta-lmp-base/recipes-support/resize-helper/resize-helper/resize-helper:67]: Another question, END here is in 512-byte sectors while we might have disks with larger sectors (e.g. 4096), is fdisk assuming 512-byte sectors by default?

**Comment by ricardosalveti** [comment]
In case you got a 4096-sector disk in hands, it would also be nice to make sure it works correctly there.

**Comment by ricardosalveti** [comment]
```
Apr 19 02:17:39 intel-corei7-64 resize-helper[411]: Warning: The kernel is still using the old partition table.
Apr 19 02:17:39 intel-corei7-64 resize-helper[411]: The new table will be used at the next reboot or after you
Apr 19 02:17:39 intel-corei7-64 resize-helper[411]: run partprobe(8) or kpartx(8)
Apr 19 02:17:39 intel-corei7-64 resize-helper[411]: The operation has completed successfully.
Apr 19 02:17:39 intel-corei7-64 resize-helper[561]: This disk is currently in use - repartitioning is probably a bad idea.
Apr 19 02:17:39 intel-corei7-64 resize-helper[561]: It's recommended to umount all file systems, and swapoff all swap
Apr 19 02:17:39 intel-corei7-64 resize-helper[561]: partitions on this disk.
Apr 19 02:17:39 intel-corei7-64 resize-helper[561]: Value out of range.
Apr 19 02:17:39 intel-corei7-64 resize-helper[561]: Partition #2 contains a ext4 signature.
Apr 19 02:17:39 intel-corei7-64 resize-helper[563]: resize2fs 1.46.4 (18-Aug-2021)
Apr 19 02:17:39 intel-corei7-64 resize-helper[563]: The filesystem is already 840751 (4k) blocks long.  Nothing to do!
```

Value out of range seems interesting. The first part is just to show that partprobe is recommended.

The output from resize2fs is not that clear to me yet.

**Comment by vanmaegima** [comment]
Tested imx7ulpea-ucom, mfgtool and boot work.

**Review by vanmaegima — APPROVED** [review]

**Review by ricardosalveti — APPROVED** [review]
LGTM, thanks a lot for working on this one.

**Pull request merged**